client/signup/steps/plans/tabbed-plans.js

```diff
@@ -0,0 +1,1316 @@
+import {
+	applyTestFiltersToPlansList,
+	findPlansKeys,
+	getPlan as getPlanFromKey,
+	getYearlyPlanByMonthly,
+	isMonthly,
+	TERM_MONTHLY,
+} from '@automattic/calypso-products';
+import { Button, Gridicon } from '@automattic/components';
+import formatCurrency from '@automattic/format-currency';
+import styled from '@emotion/styled';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { connect, useDispatch } from 'react-redux';
+import InfoPopover from 'calypso/components/info-popover';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { getPlan, getPlanBySlug, getPlanRawPrice } from 'calypso/state/plans/selectors';
+import './tabbed-plans-style.scss';
+
+function SharedFeatures( {
+	className = '',
+	selectedTab,
+	setShowFeatures,
+	sharedFeatures,
+	showFeatures,
+} ) {
+	const isMobileView = className.includes( 'mobile' );
```

Comment:
Relying on the classname is risky. Since its parent component will likely do the viewport check, I'd suggest to make it a prop. There are also chances that it can be replaced by `showFeatures` entirely; I feel that the conditions below don't look quite right, but it currently acts correctly because the `-mobile` classname is always there. Please see my comment below :)

Reply:
Ah, your tip below about isMobile is helpful! I'll use that in the refactor to clean this part up. If you're okay with it, I'll plan to leave this as-is for now since the current implementation outputs markup for both and uses CSS rules to decide which version to display.

Reply:
Cool. Let's continue to iterate on it in the follow-up PR.